src/Management/test/Endpoint.Test/Infrastructure/TestContext.cs

```diff
@@ -82,6 +82,13 @@ public T GetRequiredService<T>()
         return ServiceProvider.GetRequiredService<T>();
     }
 
+    public T GetRequiredScopedService<T>()
```

Comment:
I would recommend against that. The type isn't well-designed; it's merely a helper for what's often needed in Management tests. For example, it never disposes the service provider, which it should. Whenever we need something like this, we'd better use `WebApplicationFactory`, which was designed for replacing services in tests.

I only added this method to keep the diff minimal. It's not a great pattern either, because it doesn't work with services that implement `IDisposable`/`IAsyncDisposable`: The scope is closed before return, which makes it return a disposed instance.